### ISSUE TITLE
Inconsistent/Surprising behavior of type: ignore vs ty: ignore

### ISSUE THREAD

**Issue by TurtleOrangina** [astral-sh/ty#2767]
### Story

Consider this code:

```python
def main():
    s: str = "foo"
    i: int = 5
    s = i[3]  # type: ignore[invalid-assignment]
    print(s)
```

It gives `All checks passed!`, but what ty is doing here is *NOT* what most python programmers (or reviewers) would think is happening here. If you change the `# type: ignore[invalid-assignment]` to `# ty: ignore[invalid-assignment]`, you will get one error and one warning! This behavior violates the "Principle of least surprise". What ty is doing here is reading the `# type: ignore` and considering it as a blanket type error suppression, and ty completely ignores the rule in the brackets.

One might argue that we should simply set `respect-type-ignore-comments = false`, and teach the developers to use `ty: ignore` instead. But that also results in unwanted/unexpected results, as any "left over" `type: ignore` comments (e.g. copied in from other code bases) look like they are doing something, when in fact they are not, e.g.:

```python
def main() -> None:
    s = "foo"
    s = 5  # type: ignore[assignment]
    print(s)
```

The type: ignore was necessary when using mypy, but imagine we have switched to using only ty. `uv run ty check --config analysis.respect-type-ignore-comments=false` gives no errors. But again, the behavior of this code is surprising, as most python developers would think there is a type error being ignored here, when in reality the `type: ignore` comment should be removed for clarity.

### Suggested solution: Fully embrace # type: ignore:

- Make `# type: ignore` functionally equivalent to `# ty: ignore`, such that if a rule is specified, it needs to be a valid ty rule. This would flag things like the above mypy rule `# type: ignore[assignment]` with `ignore-comment-unknown-rule`, and allow things like suppressing only the specified ty rule via # type: ignore[invalid-argument-type]
- Users with "dual type checker setups" that like to specify the exact rules being suppressed, would set respect-type-ignore-comments = false, and then use type: ignore for e.g. mypy and ty: ignore for ty.
- Use `#type: ignore` and `#type: ignore[rule]` as the "default" ways to specify suppression (in examples/documentation). Mention `# ty: ignore` only as alternative alias that should be used in cases where multiple type checkers work on the same code base.
- In the LSP, use `# type: ignore[rule]` as offered "quick fix" for lines with a type errors if `respect-type-ignore-comments=true`. Otherwise, offer `# ty: ignore[rule]` as quick fix, as is done currently.

**Comment by carljm** [comment]
ty's behavior here follows the long-established precedent of pyright, and is also the behavior of every other non-mypy type checker which does not exactly copy mypy rule codes. So it is likely to be expected by any developer who has used any of those type checkers. The established pattern in the ecosystem for many years now is that rule codes in `type: ignore` have effectively been "claimed" by mypy.

I think it is unlikely that we will counter that precedent and take the direction proposed here; I think it's more likely that we do what is proposed in #2603.

**Comment by TurtleOrangina** [comment]
@carljm I was not aware of this precedent, all the projects I work on used mypy so far, so keeping with `type: ignore` made the most sense to me (less need to teach developers to use `ty: ignore` now). I can see a happy scenario with using `ty: ignore` instead of `type: ignore` as well, by implementing the default enabled warning suggested in #2603 .

Ideally (in addition to what is suggested in #2603) there would be an option to warn on any use `# type: ignore`, because I don't want that to be used at all in projects that use ty in this case! Something like `analysis.warn-on-type-ignore-comments=true`. Then any `# type: ignore` would be flagged and when a developer accidentally adds a type: ignore (e.g. through copy+paste), they will be asked to change it to `ty: ignore`, and if it then turns out to be unnecessary, ty will warn about that.

**Comment by carljm** [comment]
@TurtleOrangina I think that is effectively #606 -- which effectively bans all `# type: ignore` comments, since they are all blanket ignores.

**Comment by carljm** [comment]
Going to close this as a dupe of #2603, since it seems we are on the same page that the combination of #2603 and #606 are adequate here.
